system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -31,6 +33,11 @@ this.TopStoriesFeed = class TopStoriesFeed {
     this.spocsPerNewTabs = 0;
     this.newTabsSinceSpoc = 0;
     this.contentUpdateQueue = [];
+    this.dedupe = new Dedupe(this._dedupeKey);
```

Comment:
Can we pass site => site && site.url here instance of creating a pseudo-private member function?